dev/tools/coqdev.el

```diff
@@ -78,11 +78,7 @@ Specifically `camldebug-command-name' and `ocamldebug-command-name'."
 Note that this function is executed before _Coqproject is read if it exists."
   (let ((dir (coqdev-default-directory)))
     (when dir
-      (unless coq-prog-args
-        (setq coq-prog-args
-              `("-coqlib" ,dir
-                "-topfile" ,buffer-file-name)))
-      (setq-local coq-prog-name (concat dir "bin/coqtop")))))
+      (setq-local coq-prog-name (concat dir "_build/default/dev/shim/coqtop-prelude")))))
```

Comment:
Umm, you don't want to make this conditional on the existence of the `_build` dir? Also, should you use `dune exec -- dev/shim/coqtop-prelude`, that'll make sure the build is up to date, but maybe you don't want that?

Reply:
>Umm, you don't want to make this conditional on the existence of the _build dir?

No point for me. I have proofgeneral ask me before running the program so if there's a problem I can type some other path in. I don't know what other people's setup is like so can't comment.

>Also, should you use dune exec -- dev/shim/coqtop-prelude, that'll make sure the build is up to date, but maybe you don't want that?

Definitely don't want that.

Reply:
A thing to take into account is that the planned way to run `coqtop` under dune will be `dune coqtop file`, but that invocation will indeed make sure that the dependencies of `file` are up to date.

Reply:
You can also use `dune exec --no-build -- dev/shim/coqtop-prelude` if you don't want to rebuild. The point of preferring exec is that it will setup the env vars correctly for you.

Reply:
Indeed, maybe we should make it a flag @SkySkimmer ?

Reply:
I like having it be the same as what I run in unrelated directories (I often run coqtop/PG on code I put in /tmp/foo.v for short experiments (eg gitter quizes ;)))
Also, calling dune means having `coq-prog-name` be just `dune` and passing the rest via `coq-prog-args`, not sure how well that would work in practice.

>Indeed, maybe we should make it a flag @SkySkimmer ?

Not sure what you mean, what would be a flag of what?

Reply:
I mean introducing a flag for whether to rebuild. In general, I think that in most cases `dune coqtop Foo.v` should rebuild dependencies of `Foo`, that's what's done for OCaml and what modern IDEs do, and it makes sense IMHO.

It is frustrating to have to remember to type make manually I think.

We could have a flag to disable that.

Reply:
Isn't that already in the roadmap? I thought I read about that idea somewhere